apps/aecore/src/aec_tx_sign.erl

```diff
@@ -147,11 +148,30 @@ serialize_for_client(Encoding, Header, #signed_tx{}=S) ->
 
 serialize_for_client(message_pack, #signed_tx{}=S, BlockHeight, BlockHash) ->
     GenericData = [#{<<"block_height">> => BlockHeight},
-                   #{<<"block_hash">> => BlockHash}],
-    aec_base58c:encode(transaction, msgpack:pack(serialize(S) ++ GenericData));
+                   #{<<"block_hash">> => aec_base58c:encode(block_hash, BlockHash)}],
+    aec_base58c:encode(transaction, msgpack:pack(serialize(S) ++ [GenericData]));
 serialize_for_client(json, #signed_tx{data = Tx, signatures = Sigs},
                      BlockHeight, BlockHash) ->
     #{<<"tx">> => aec_tx:serialize_for_client(Tx),
       <<"block_height">> => BlockHeight,
       <<"block_hash">> => aec_base58c:encode(block_hash, BlockHash),
       <<"signatures">> => lists:map(fun(Sig) -> aec_base58c:encode(signature, Sig) end, Sigs)}. 
+
+extract_encoded_generic_data(message_pack, Bin) ->
```

Comment:
As a reader I do not easily correlate `extract_encoded_generic_data` with `serialize_for_client` - closely coupling is implicit. I would suggest either enriching name of `extract_encoded_generic_data`, or (better) write type spec to `extract_encoded_generic_data` and `serialize_for_client` so that it is clear they operate on same data.